components/ledger/ledger-common-flow/src/main/kotlin/net/corda/ledger/common/flow/impl/transaction/TransactionSignatureServiceImpl.kt

```diff
@@ -257,10 +280,8 @@ class TransactionSignatureServiceImpl @Activate constructor(
         }
     }
 
-    private fun checkAndGetSignatureSpec(signatureWithMetadata: DigitalSignatureAndMetadata): SignatureSpec {
-        val signatureSpec = signatureWithMetadata.metadata.signatureSpec
-
-        val compatibleSpecs = signatureSpecService.compatibleSignatureSpecs(signatureWithMetadata.by)
+    private fun checkAndGetSignatureSpec(signatureSpec: SignatureSpec, signingKey: PublicKey): SignatureSpec {
```

Comment:
This one looks like checking only the signature spec now.
```suggestion
    private fun checkSignatureSpec(signatureSpec: SignatureSpec, signingKey: PublicKey): SignatureSpec {
```
Maybe?

Reply:
I think it still also returns the `SignatureSpec` though?

Reply:
Yes, it returns the signatureSpec, but you have to feed it in, and it just passes it through when the check works, so it does not get the signature spec for you. I think the renaming proposal is correct.

Reply:
Ah yes indeed, change will be applied.